src/rust/engine/process_execution/src/remote.rs

```diff
@@ -934,12 +972,49 @@ pub async fn make_execute_request(
   req: &Process,
   instance_name: Option<String>,
   cache_key_gen_version: Option<String>,
-  _store: &Store,
+  store: &Store,
+  append_only_caches_base_path: Option<&str>,
 ) -> Result<EntireExecuteRequest, String> {
+  const CACHES_WRAPPER: &str = "./__wrapper__";
+
+  // Implement append-only caches by running a wrapper script before the actual
+
+  let wrapper_script_digest_opt = match (append_only_caches_base_path, &req.append_only_caches) {
+    (Some(base_path), caches) if !caches.is_empty() => {
+      if req.working_directory.is_some() {
+        return Err(
+          "TODO-DEV: Append-only caches and working directory cannot be set together in remote execution.".into()
+        );
+      }
```

Comment:
I suspect this may cause some actual errors for Pants. But I guess we could always whack-a-mole. Instead, thoughts on simply dynamically disabling remote cache? Maybe log at debug level a warning

Reply:
I should just have the wrapper script account for the working directory when it is setting up the append-only caches.

Reply:
This is implemented now.